### PR TITLE
Switch to test helper resources operations

### PR DESCRIPTION
**Pull request by bogdando**
Use conditions, statefulsets, namespaces, jobs basic test ops
from test helper common lib

Consume ConditionGetterFunc from common lib.

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/131

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -91,4 +91,6 @@ require (
 
 replace github.com/openstack-k8s-operators/nova-operator/api => ./api
 
-replace github.com/openstack-k8s-operators/lib-common/modules/common => github.com/gibizer/lib-common/modules/common v0.0.0-20221219140315-47b042183a5b
+replace github.com/openstack-k8s-operators/lib-common/modules/common => /opt/go/src/github.com/openstack-k8s-operators/lib-common/modules/common
+
+replace github.com/openstack-k8s-operators/lib-common/modules/test => /opt/go/src/github.com/openstack-k8s-operators/lib-common/modules/test
```
Comment (bogdando) [inline go.mod:96]: DNM, those are to go once the dependency merged

File: test/functional/suite_test.go [hunk 2]
```diff
@@ -126,10 +126,12 @@ var _ = BeforeSuite(func() {
 
 	//+kubebuilder:scaffold:scheme
 
+	logger = ctrl.Log.WithName("---Test---")
```
Comment (bogdando) [inline test/functional/suite_test.go:129]: this is how operators can override the testhelper logger instance

**Review by gibizer — CHANGES_REQUESTED** [review]
it needs a rebase but otherwise looks good.

File: controllers/nova_controller.go [hunk 3]
```diff
@@ -26,12 +26,12 @@ import (
 	"k8s.io/apimachinery/pkg/types"
```
[the remark this answers is not in the document]
Comment (bogdando) [reply controllers/nova_controller.go]: go mod tidy in the rebased branch doesn't make this to vanish

**Comment by bogdando** [comment]
> it needs a rebase but otherwise looks good.

done

**Review by gibizer — APPROVED** [review]
looks good. thanks!

**Pull request merged**